dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java

```diff
@@ -903,7 +903,8 @@ protected int createComplementDependentCommand(List<Schedule> schedules, Command
         List<DependentProcessDefinition> dependentProcessDefinitionList =
                 getComplementDependentDefinitionList(dependentCommand.getProcessDefinitionCode(),
                         CronUtils.getMaxCycle(schedules.get(0).getCrontab()), dependentCommand.getWorkerGroup());
-
+        // If the id is Integer, the auto-increment id will be obtained and cloned, causing duplicate writes
```

Comment:
How did you get this conclusion? From the documentation of `BeanUtils.cloneBean`, it copies all fields regardless of its type

Reply:
Before using int, mybtais-plus did not fill in the id field, it default use 0, but after being changed to Integer, it will fill it for auto-increment id.

Reply:
> Before using int, mybtais-plus did not fill in the id field, it default use 0, but after being changed to Integer, it will fill it for auto-increment id.

OK I see, this is a really strange behavior for mybatis-plus...